src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,181 @@
+        /// <param name="pwmChannel">The PWM channel number.</param>
+        /// <param name="frequency">The frequency in hertz.</param>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage represented as a value between 0.0 and 1.0.</param>
+        /// <param name="polarity">The polarity the channel should start in.</param>
+        public UnixPwmChannel(
+            int pwmChip,
+            int pwmChannel,
+            double frequency,
+            double dutyCyclePercentage = 0.5,
+            PwmPolarity polarity = PwmPolarity.ActiveHigh)
+        {
+            _pwmChip = pwmChip;
+            _pwmChannel = pwmChannel;
+            ValidatePwmChannel();
+            OpenChannel();
+            Thread.Sleep(100);  // TODO: Need a better solution to delay for available file.
+
+            SetFrequency(frequency);
+            DutyCyclePercentage = dutyCyclePercentage;
+
+            // NOTE: There appears to be a lower-level bug that won't allow the
+            // polarity to be changed until a valid period has been written.
+            SetPolarity(polarity);
+        }
+
+        /// <summary>
+        /// The frequency in hertz.
+        /// </summary>
+        public override double Frequency => _frequency;
+
+        /// <summary>
+        /// The duty cycle percentage represented as a value between 0.0 and 1.0.
+        /// </summary>
+        public override double DutyCyclePercentage
+        {
+            get
+            {
+                return _dutyCyclePercentage;
+            }
+            set
+            {
+                SetDutyCyclePercentage(value);
+            }
+        }
+
+        // Helpers for file paths.
+        private string GetChipPath() => $"/sys/class/pwm/pwmchip{_pwmChip}";
+        private string GetChannelPath() => $"{GetChipPath()}/pwm{_pwmChannel}";
+        private string GetEnablePath() => $"{GetChannelPath()}/enable";
+
+        private void ValidatePwmChannel()
+        {
+            if (!Directory.Exists(GetChipPath()))
+            {
+                throw new ArgumentException($"The chip number {_pwmChip} is invalid or is not enabled.");
+            }
+
+            string npwmPath = $"{GetChipPath()}/npwm";
+
+            if (int.TryParse(File.ReadAllText(npwmPath), out int numberOfSupportedChannels))
+            {
+                if (_pwmChannel < 0 || _pwmChannel >= numberOfSupportedChannels)
+                {
+                    throw new ArgumentException($"The PWM chip {_pwmChip} does not support the channel {_pwmChannel}.");
+                }
+            }
+            else
+            {
+                throw new IOException($"Unable to parse the number of supported channels at {npwmPath}.");
+            }
+        }
+
+        // In Linux, the period needs to be a whole number and can't have a decimal point.
+        private int GetPeriodInNanoSeconds(double frequency) => (int)((1.0 / frequency) * 1_000_000_000);
```

Comment:
would it worked if we changed code to use Period instead of frequency to avoid using fractions? Or perhaps just change Frequency to be integer Hz - is there any advantage of using double?

Reply:
> is there any advantage of using double?

No, I used since it was used in original code.  I can review/change.  I like `frequency` better as most peeps will understand it versus `period`.

Reply:
Updated in next commit.